lib/aptible/cli/helpers/s3_log_helpers.rb

```diff
@@ -0,0 +1,218 @@
+require 'aws-sdk-s3'
+require 'pathname'
+
+module Aptible
+  module CLI
+    module Helpers
+      module S3LogHelpers
+        def ensure_aws_creds
+          cred_errors = []
+          unless ENV['AWS_ACCESS_KEY_ID']
+            cred_errors << 'Missing environment variable: AWS_ACCESS_KEY_ID'
+          end
+          unless ENV['AWS_SECRET_ACCESS_KEY']
+            cred_errors << 'Missing environment variable: AWS_SECRET_ACCESS_KEY'
+          end
+          raise Thor::Error, cred_errors.join(' ') if cred_errors.any?
+        end
+
+        def validate_log_search_options(options = {})
+          id_options = [
+            options[:app_id],
+            options[:database_id],
+            options[:endpoint_id],
+            options[:container_id]
+          ]
+          date_options = [options[:start_date], options[:end_date]]
+          unless options[:string_matches] || id_options.any?
+            m = 'You must specify an option to identify the logs to download,' \
+                ' either: --string-matches, --app-id, --database-id,' \
+                ' --endpoint-id, or --container-id'
+            raise Thor::Error, m
+          end
+
+          m = 'You cannot pass --app-id, --database-id, --endpoint-id, or ' \
+              '--container-id when using --string-matches.'
+          raise Thor::Error, m if options[:string_matches] && id_options.any?
+
+          m = 'You must specify only one of ' \
+              '--app-id, --database-id, --endpoint-id or --container-id'
+          raise Thor::Error, m if id_options.any? && !id_options.one?
+
+          m = 'The options --start-date/--end-date cannot be used when ' \
+              'searching by string'
+          raise Thor::Error, m if options[:string_matches] && date_options.any?
```

Comment:
Why not?

Reply:
No reason, other than i wrote the string search first for the v2 format, and then added support for v3 format that includes the date. There's no design requirements here so 🤷 maybe a future improvement?

Reply:
I suppose the main reason to use `--string-matches` is to query times in v2. 👍